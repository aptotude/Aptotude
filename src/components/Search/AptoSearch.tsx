import React from 'react';
import './aptoSearch.scss';
import { Search, SearchProps } from 'semantic-ui-react';
import { AptoIcon } from '../Icon';
import { searchIcon } from './SearchIcon';
import { AptoSpinner } from '../Spinner';

export class AptoSearch extends React.Component<SearchProps> {
  public render() {
    const { loading, ...rest } = this.props;
    return (
      <div className="AptoSearch">
        <Search input={{ icon: null }} {...rest} />
        {!loading && <AptoIcon icon={searchIcon} />}
        {loading && <AptoSpinner noOverlay={true} />}
      </div>
    );
  }
}
