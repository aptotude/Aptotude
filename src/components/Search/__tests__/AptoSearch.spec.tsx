import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoSearch } from '../AptoSearch';

afterEach(() => {
  cleanup();
});

describe('AptoSearch', () => {
  it('Renders Component', () => {
    const { container } = render(<AptoSearch />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders Component with name/value', () => {
    const { container } = render(<AptoSearch name="hi" value="woo" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders loading', () => {
    const { container } = render(<AptoSearch loading />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders disabled', () => {
    const { container } = render(<AptoSearch disabled />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders results', () => {
    const results = [{ title: 'foo', value: 'bar' }];

    const { container } = render(<AptoSearch results={results} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders result categories', () => {
    const results = {
      things: {
        name: 'things',
        results: [{ title: 'foo', value: 'bar' }]
      },
      stuff: {
        name: 'stuff',
        results: [{ title: 'boo', value: 'car' }]
      }
    };

    const { container } = render(<AptoSearch category results={results} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
