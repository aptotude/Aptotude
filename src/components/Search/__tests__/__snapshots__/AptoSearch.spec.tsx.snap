// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoSearch Renders Component 1`] = `
<div
  class="AptoSearch"
>
  <div
    class="ui search"
  >
    <div
      class="ui input"
    >
      <input
        autocomplete="off"
        class="prompt"
        tabindex="0"
        type="text"
        value=""
      />
    </div>
    <div
      class="results transition"
    >
      <div
        class="message empty"
      >
        <div
          class="header"
        >
          No results found.
        </div>
      </div>
    </div>
  </div>
  <span
    class="AptoIcon AptoIcon--size3"
  >
    <svg
      focusable="false"
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 24 24"
      width="100%"
    >
      <path
        d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0 0 16 9.5 6.5 6.5 0 1 0 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
</div>
`;

exports[`AptoSearch Renders Component with name/value 1`] = `
<div
  class="AptoSearch"
>
  <div
    class="ui search"
  >
    <div
      class="ui input"
    >
      <input
        autocomplete="off"
        class="prompt"
        name="hi"
        tabindex="0"
        type="text"
        value="woo"
      />
    </div>
    <div
      class="results transition"
    >
      <div
        class="message empty"
      >
        <div
          class="header"
        >
          No results found.
        </div>
      </div>
    </div>
  </div>
  <span
    class="AptoIcon AptoIcon--size3"
  >
    <svg
      focusable="false"
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 24 24"
      width="100%"
    >
      <path
        d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0 0 16 9.5 6.5 6.5 0 1 0 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
</div>
`;

exports[`AptoSearch Renders disabled 1`] = `
<div
  class="AptoSearch"
>
  <div
    class="ui search"
  >
    <div
      class="ui disabled input"
    >
      <input
        autocomplete="off"
        class="prompt"
        disabled=""
        tabindex="0"
        type="text"
        value=""
      />
    </div>
    <div
      class="results transition"
    >
      <div
        class="message empty"
      >
        <div
          class="header"
        >
          No results found.
        </div>
      </div>
    </div>
  </div>
  <span
    class="AptoIcon AptoIcon--size3"
  >
    <svg
      focusable="false"
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 24 24"
      width="100%"
    >
      <path
        d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0 0 16 9.5 6.5 6.5 0 1 0 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
</div>
`;

exports[`AptoSearch Renders loading 1`] = `
<div
  class="AptoSearch"
>
  <div
    class="ui search"
  >
    <div
      class="ui input"
    >
      <input
        autocomplete="off"
        class="prompt"
        tabindex="0"
        type="text"
        value=""
      />
    </div>
    <div
      class="results transition"
    >
      <div
        class="message empty"
      >
        <div
          class="header"
        >
          No results found.
        </div>
      </div>
    </div>
  </div>
  <div
    class="AptoSpinner AptoSpinner--noOverlay"
  >
    <div
      class="AptoSpinner-spinner"
    >
      <svg
        viewBox="0 0 64 64"
      >
        <circle
          r="26"
          transform="translate(32,32)"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`AptoSearch Renders result categories 1`] = `
<div
  class="AptoSearch"
>
  <div
    class="ui category search"
  >
    <div
      class="ui input"
    >
      <input
        autocomplete="off"
        class="prompt"
        tabindex="0"
        type="text"
        value=""
      />
    </div>
    <div
      class="results transition"
    >
      <div
        class="category"
      >
        <div
          class="name"
        >
          things
        </div>
        <div
          class="results"
        >
          <div
            class="result"
            value="bar"
          >
            <div
              class="content"
            >
              <div
                class="title"
              >
                foo
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="category"
      >
        <div
          class="name"
        >
          stuff
        </div>
        <div
          class="results"
        >
          <div
            class="result"
            value="car"
          >
            <div
              class="content"
            >
              <div
                class="title"
              >
                boo
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    class="AptoIcon AptoIcon--size3"
  >
    <svg
      focusable="false"
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 24 24"
      width="100%"
    >
      <path
        d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0 0 16 9.5 6.5 6.5 0 1 0 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
</div>
`;

exports[`AptoSearch Renders results 1`] = `
<div
  class="AptoSearch"
>
  <div
    class="ui search"
  >
    <div
      class="ui input"
    >
      <input
        autocomplete="off"
        class="prompt"
        tabindex="0"
        type="text"
        value=""
      />
    </div>
    <div
      class="results transition"
    >
      <div
        class="result"
        value="bar"
      >
        <div
          class="content"
        >
          <div
            class="title"
          >
            foo
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    class="AptoIcon AptoIcon--size3"
  >
    <svg
      focusable="false"
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 24 24"
      width="100%"
    >
      <path
        d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0 0 16 9.5 6.5 6.5 0 1 0 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
</div>
`;
